sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/ParserUtils.scala

```diff
@@ -259,4 +259,48 @@ object ParserUtils {
       }
     }
   }
+
+  private def stripGap(sb: StringBuilder): Unit = {
+    if (sb.length > 0 && sb.charAt(sb.length - 1) == ' ') {
+      sb.setLength(sb.length - 1)
+    }
+  }
+
+  /**
+   * Adds a gap after every term node except of '(', ')', '[', ']',
+   * and removes a gap before '(', ')', '[', ']', ','.
+   */
+  private def toExprAlias(ctx: ParseTree, sb: StringBuilder): Unit = {
+    val childCount = ctx.getChildCount
+    if (childCount > 0) {
```

Comment:
seems we can remove the `if`, as the while loop is noop if count is 0.

Reply:
removed